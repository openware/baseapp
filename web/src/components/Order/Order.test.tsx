import { mount, shallow, ShallowWrapper } from 'enzyme';
import * as React from 'react';
import { SinonSpy, spy } from 'sinon';
import { Order, OrderComponentProps } from './';

// tslint:disable:no-magic-numbers

const defaultProps: OrderComponentProps = {
    onSubmit: spy(),
    priceMarketBuy: 5,
    priceMarketSell: 10,
    currentMarketAskPrecision: 4,
    currentMarketBidPrecision: 5,
    availableBase: 200,
    availableQuote: 12,
    from: 'btc',
    to: 'eth',
    asks: [['10','1']],
    bids: [['10','1']],
    currentMarketFilters: [],
    translate: jest.fn(),
};

const setup = (props: Partial<OrderComponentProps> = {}) =>
    shallow(<Order {...{ ...defaultProps, ...props }} />);

describe('Order', () => {
    let wrapper: ShallowWrapper;

    beforeEach(() => {
        wrapper = setup();
    });

    it('should match snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('button should be disabled', () => {
        // tslint:disable: no-shadowed-variable
        const wrapper = mount(<Order {...{ ...defaultProps}} />);
        wrapper.find('input').at(2).simulate('click');
        expect((defaultProps.onSubmit as SinonSpy).calledOnceWith()).toBeFalsy();
        wrapper.unmount();
    });
});
