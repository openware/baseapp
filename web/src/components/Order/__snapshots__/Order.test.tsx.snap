// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order should match snapshot 1`] = `
<div
  className="cr-order cr-order--extended"
>
  <div
    className="cr-order--extended__buy"
  >
    <Component
      currentTabIndex={0}
      fixed={true}
      onTabChange={[Function]}
      panels={
        Array [
          Object {
            "content": <OrderForm
              amount=""
              available={12}
              currentMarketAskPrecision={4}
              currentMarketBidPrecision={5}
              currentMarketFilters={Array []}
              from="btc"
              handleAmountChange={[Function]}
              handleChangeAmountByButton={[Function]}
              onSubmit={[Function]}
              orderTypes={
                Array [
                  "Limit",
                  "Market",
                ]
              }
              orderTypesIndex={
                Array [
                  "Limit",
                  "Market",
                ]
              }
              priceMarket={5}
              to="eth"
              totalPrice={0}
              translate={
                [MockFunction] {
                  "calls": Array [
                    Array [
                      "page.body.trade.header.newOrder.content.tabs.buy",
                    ],
                    Array [
                      "page.body.trade.header.newOrder.content.tabs.sell",
                    ],
                  ],
                  "results": Array [
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                  ],
                }
              }
              type="buy"
            />,
            "label": "Buy",
          },
        ]
      }
    />
  </div>
  <div
    className="cr-order--extended__sell"
  >
    <Component
      currentTabIndex={0}
      fixed={true}
      onTabChange={[Function]}
      panels={
        Array [
          Object {
            "content": <OrderForm
              amount=""
              available={200}
              currentMarketAskPrecision={4}
              currentMarketBidPrecision={5}
              currentMarketFilters={Array []}
              from="btc"
              handleAmountChange={[Function]}
              handleChangeAmountByButton={[Function]}
              onSubmit={[Function]}
              orderTypes={
                Array [
                  "Limit",
                  "Market",
                ]
              }
              orderTypesIndex={
                Array [
                  "Limit",
                  "Market",
                ]
              }
              priceMarket={10}
              to="eth"
              totalPrice={0}
              translate={
                [MockFunction] {
                  "calls": Array [
                    Array [
                      "page.body.trade.header.newOrder.content.tabs.buy",
                    ],
                    Array [
                      "page.body.trade.header.newOrder.content.tabs.sell",
                    ],
                  ],
                  "results": Array [
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                  ],
                }
              }
              type="sell"
            />,
            "label": "Sell",
          },
        ]
      }
    />
  </div>
</div>
`;
